packages/eslint-mdx/src/helpers.ts

```diff
@@ -146,16 +146,17 @@ export const restoreNodeLocation = <T>(
     return node
   }
 
-  const { range } = node
+  let {
+    range: [start, end],
+  } = node
 
-  const start = range[0] + offset
-  const end = range[1] + offset
+  const range = [(start += offset), (end += +offset)] as const
```

Comment:
The extra `+` looks like a typo?

```suggestion
  const range = [(start += offset), (end += offset)] as const
```

Reply:
Done

Reply:
raised a new rule proposal for it https://github.com/typescript-eslint/typescript-eslint/issues/3204